src/read.jl

```diff
@@ -470,14 +519,45 @@ function reformat_demand_flexibility_input(
 
             # add header and datetime index column
             DataFrames.rename!(eq_bus_df, Symbol.(case.busid[flex_bus_idx]))
-            DataFrames.insertcols!(eq_bus_df, 1, :"UTC Time" => fh["UTC Time"])
+            DataFrames.insertcols!(
+                eq_bus_df, 1, :"UTC Time" => demand_flex_field["UTC Time"]
+            )
+
+            # store to new df object
+            demand_flexibility_updated[field] = eq_bus_df
+
+            # if flex amount specified in DOE data, re-format using bus EIA ID
+        elseif demand_flexibility.enable_doe_flexibility &&
+            field in ["flex_amt_up", "flex_amt_dn"]
+
+            # all load buses with valid EIA ID 
+            flexible_bus_idx = intersect(sets.load_bus_idx, findall(case.bus_eiaid .> 0))
+            flexible_bus_num = length(flexible_bus_idx)
+
+            # bus flexibility Percentage matrix
+            doe_bus_flexibility = zeros(size(case.demand, 1), flexible_bus_num)
+            doe_eiaid = [parse(Int, x) for x in names(demand_flex_field)[2:end]]
+            for i in 1:flexible_bus_num
+                if case.bus_eiaid[flexible_bus_idx[i]] in doe_eiaid
+                    doe_bus_flexibility[:, i] = demand_flex_field[Symbol(
+                        case.bus_eiaid[flexible_bus_idx[i]]
+                    )]
+                end
+            end
```

Comment:
I guess the if check might be slow when it checks if the eiaid of each bus is valid. Maybe this can be removed as the check is already done when assigning the eiaid to buses in prereise.

Reply:
Since we are providing that data and the check is already done in `PreREISE`, I would agree that having the check in `REISE.jl` is probably not necessary. @BainanXia, what are your thoughts? Might a check like this still be necessary if users are adding extra buses? Or would this check make more sense in `PowerSimData` (once we add this to `PowerSimData`) for new buses?

Reply:
I just wanted to follow-up on my previous comment here. Thoughts?